### PR TITLE
AbsPkt improvements

### PR DESCRIPTION
**Pull request by mlimbeck**
- moved `absInfofield` functions from `hopfield_spec.gobra`to` infofield_spec.gobra`
- added opaque to `absPkt` and subsequent functions

### CODE REVIEW

File: pkg/slayers/path/hopfield_spec.gobra
Comment (jcp19) [inline pkg/slayers/path/hopfield_spec.gobra:44]: Nice catch! Thanks for moving this!

File: router/io-spec.gobra
Comment (jcp19) [inline router/io-spec.gobra:484]: This is a good idea

File: router/io-spec.gobra
Comment (jcp19) [inline router/io-spec.gobra:486]: Why are these posts here? Is the goal to prove them before we merge? Maybe sth that might make sense here is to prove these conditions extrinsically (i.e., on a separate lemma) and call the lemma when we need them.

**Review by jcp19 — APPROVED** [review]
This looks all fine to me. But didn't we talk about merging this in master instead?

File: router/io-spec.gobra
Comment (mlimbeck) [reply router/io-spec.gobra]: Yes if we want to keep everything opaque, we have to prove them separately. I removed them for now.

**Review by Dspil — APPROVED** [review]

**Comment by jcp19** [comment]
it seems that this PR introduced a new verification error. @mlimbeck can you fix it before we merge?

**Comment by jcp19** [comment]
Interestingly, the last commit performs changes in the same lines as PR #276, which @Dspil introduced so that he could make progress with the unsupported paths. I guess both postconditions make sense, but we should discuss briefly over a meeting which ones are the most useful. We can merge this PR after we clarify that, hopefully tomorrow.

File: router/io-spec.gobra
Comment (jcp19) [inline router/io-spec.gobra:196]: could we have the postconditions proven in a separate lemma?

File: router/io-spec.gobra
Comment (jcp19) [inline router/io-spec.gobra:247]: same here

File: router/io-spec.gobra
Comment (jcp19) [inline router/io-spec.gobra:485]: as a general guideline, we should put the `opaque` modifier in the line right after `ghost` to make it easier to read.

**Review by jcp19 — APPROVED** [review]

File: router/io-spec.gobra
Comment (mlimbeck) [reply router/io-spec.gobra]: Okay, I changed it.

**Pull request merged**
